mmdet/models/task_modules/samplers/sampling_result.py

```diff
@@ -84,60 +82,118 @@ class SamplingResult(util_mixins.NiceRepr):
     """
 
     def __init__(self,
+                 pred_instances: InstanceData,
+                 gt_instances: InstanceData,
                  pos_inds: Tensor,
                  neg_inds: Tensor,
-                 priors: Tensor,
-                 gt_bboxes: Tensor,
-                 assign_result: AssignResult,
-                 gt_flags: Tensor,
                  avg_factor_with_neg: bool = True) -> None:
+        self.pred_instances = pred_instances
+        self.gt_instances = gt_instances
+        self.pos_inds = pos_inds
+        self.neg_inds = neg_inds
+        self.num_pos = max(pos_inds.numel(), 1)
+        self.num_neg = max(neg_inds.numel(), 1)
+        self.avg_factor_with_neg = avg_factor_with_neg
+        self.avg_factor = self.num_pos + self.num_neg \
+            if avg_factor_with_neg else self.num_pos
+
+        self.pos_assigned_gt_inds = pred_instances.gt_inds[pos_inds] - 1
+
+    def update_sampling_result(self,
```

Comment:
This logic should be placed in the sampler, instead of in a method of the AssignResult.